mayastor/src/lvm/lvm.rs

```diff
@@ -0,0 +1,285 @@
+        None => Err(Error::NotFound {err: format!("lvm volume {} not found", req.name)}),
+    }
+}
+
+pub async fn create_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let output = Command::new(PVCREATE_COMMAND)
+        .args(&req.disks)
+        .output()
+        .await
+        .expect("failed to execute pv_create");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to parse stderr for pv_create".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg})
+    }
+
+    let output = Command::new(VGCREATE_COMMAND)
+        .arg(req.name.as_str())
+        .args(&req.disks)
+        .arg("--addtag=mayastor")
+        .output()
+        .await
+        .expect("failed to execute vg_create");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to execute vg_create".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+
+    get_lvm_vg(req.name).await
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct VGListReport {
+    report: Vec<VGList>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VGList {
+    vg: Vec<VGName>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VGName {
+    vg_name: String
+}
+pub async fn list_lvm_vg() -> Result<Vec<Pool>, Error> {
+    let output = Command::new(VGS_COMMAND)
+        .args(&["--options=vg_name", "--reportformat=json"])
+        .arg("@mayastor")
+        .output()
+        .await
+        .expect("failed to execute vgs");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to execute vgs".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+
+    let r: VGListReport = match serde_json::from_slice(output.stdout.as_slice()){
+        Ok(r) => r,
+        Err(e) => return Err(Error::FailedParsing{err: e.to_string()}),
+    };
+
+    let mut res: Vec<Pool> = vec![];
+
+    for v in r.report[0].vg.as_slice() {
+        let pool = match get_lvm_vg(v.vg_name.clone()).await {
+            Ok(p) => p,
+            Err(e) => return Err(e),
+        };
+        res.push(pool);
+    }
+
+    Ok(res)
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct VGsReport {
+    report: Vec<VG>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VG {
+    vg: Vec<VGSize>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VGSize {
+    vg_size: String,
+    vg_free: String
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct PVsReport {
+    report: Vec<PV>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct PV {
+    pv: Vec<VgPvMap>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VgPvMap {
+    vg_name: String,
+    pv_name: String
+}
+
+pub async fn get_lvm_vg(name: String) -> Result<Pool, Error> {
```

Comment:
IIRC we use verbs like list rather than get. Get is used to get properties of a volume instance but not to get all the volumes, dont we?

Reply:
This get function is used to call the vgs and pvs commands to find out the capacity, used & disks for a vol group and convert that into a rpc::Pool struct, so I guess get makes more sense here